src/components/shoppingList/shoppingList.tsx

```diff
@@ -0,0 +1,211 @@
+import {
+  useState,
+  useRef,
+  useEffect,
+  type MouseEventHandler,
+  type FormEventHandler,
+  type ReactElement,
+  type CSSProperties,
+} from 'react'
+import AnimateHeight from 'react-animate-height'
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import { faXmark, faPenToSquare } from '@fortawesome/free-solid-svg-icons'
+import { type RequestShoppingList } from '../../types/apiData'
+import {
+  useColorScheme,
+  usePageContext,
+  useShoppingListsContext,
+} from '../../hooks/contexts'
+import useComponentVisible from '../../hooks/useComponentVisible'
+import useSize from '../../hooks/useSize'
+import ShoppingListItemCreateForm from '../shoppingListItemCreateForm/shoppingListItemCreateForm'
+import ListEditForm from '../listEditForm/listEditForm'
+import styles from './shoppingList.module.css'
+
+interface ShoppingListProps {
+  listId: number
+  title: string
+  canEdit?: boolean
+  children?: ReactElement | ReactElement[] | null
+}
+
+const ShoppingList = ({
+  listId,
+  title,
+  canEdit = false,
+  children,
+}: ShoppingListProps) => {
+  const DELETE_CONFIRMATION = `Are you sure you want to delete the list "${title}"? You will also lose any list items on the list. This action cannot be undone.`
```

Comment:
This is only used in one place so should probably be moved closer to that place.